File: VB/Q205267_4/PersistentClasses1.vb
Imports DevExpress.Xpo

Namespace Northwind

    Public Class Employees
        Inherits XPLiteObject

        Private fEmployeeID As Integer

        <Key(True)>
        Public Property EmployeeID As Integer
            Get
                Return fEmployeeID
            End Get

            Set(ByVal value As Integer)
                SetPropertyValue(Of Integer)("EmployeeID", fEmployeeID, value)
            End Set
        End Property

        Private fFirstName As String

        <Size(10)>
        Public Property FirstName As String
            Get
                Return fFirstName
            End Get

            Set(ByVal value As String)
                SetPropertyValue("FirstName", fFirstName, value)
            End Set
        End Property

        <Association("Employee-Orders", GetType(Orders))>
        Public ReadOnly Property Orders As XPCollection(Of Orders)
            Get
                Return GetCollection(Of Orders)("Orders")
            End Get
        End Property

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Sub New()
            MyBase.New(Session.DefaultSession)
        End Sub

        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
        End Sub
    End Class

    Public Class Orders
        Inherits XPLiteObject

        Private fOrderID As Integer

        <Key(True)>
        Public Property OrderID As Integer
            Get
                Return fOrderID
            End Get

            Set(ByVal value As Integer)
                SetPropertyValue(Of Integer)("OrderID", fOrderID, value)
            End Set
        End Property

        Private fEmployeeID As Employees

        <Association("Employee-Orders")>
        Public Property EmployeeID As Employees
            Get
                Return fEmployeeID
            End Get

            Set(ByVal value As Employees)
                SetPropertyValue("EmployeeID", fEmployeeID, value)
            End Set
        End Property

        Private fShipCity As String

        <Size(15)>
        Public Property ShipCity As String
            Get
                Return fShipCity
            End Get

            Set(ByVal value As String)
                SetPropertyValue("ShipCity", fShipCity, value)
            End Set
        End Property

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Sub New()
            MyBase.New(Session.DefaultSession)
        End Sub

        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
        End Sub
    End Class
End Namespace

File: CS/Q205267_4/PersistentClasses1.cs
using System;
using DevExpress.Xpo;
namespace Northwind {

    public class Employees : XPLiteObject {
        int fEmployeeID;
        [Key(true)]
        public int EmployeeID {
            get { return fEmployeeID; }
            set { SetPropertyValue<int>("EmployeeID", ref fEmployeeID, value); }
        }
        string fFirstName;
        [Size(10)]
        public string FirstName {
            get { return fFirstName; }
            set { SetPropertyValue<string>("FirstName", ref fFirstName, value); }
        }
        [Association("Employee-Orders", typeof(Orders))]
        public XPCollection<Orders> Orders { get { return GetCollection<Orders>("Orders"); } }
        public Employees(Session session) : base(session) { }
        public Employees() : base(Session.DefaultSession) { }
        public override void AfterConstruction() { base.AfterConstruction(); }
    }
    public class Orders : XPLiteObject {
        int fOrderID;
        [Key(true)]
        public int OrderID {
            get { return fOrderID; }
            set { SetPropertyValue<int>("OrderID", ref fOrderID, value); }
        }
        Employees fEmployeeID;
        [Association("Employee-Orders")]
        public Employees EmployeeID {
            get { return fEmployeeID; }
            set { SetPropertyValue<Employees>("EmployeeID", ref fEmployeeID, value); }
        }
        string fShipCity;
        [Size(15)]
        public string ShipCity {
            get { return fShipCity; }
            set { SetPropertyValue<string>("ShipCity", ref fShipCity, value); }
        }
        public Orders(Session session) : base(session) { }
        public Orders() : base(Session.DefaultSession) { }
        public override void AfterConstruction() { base.AfterConstruction(); }
    }
}